File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.c
#include <stdio.h>
#include <string.h>

int main(void) { 
   char str[21];
   
   while(1){
      int flag = 0, vowel = 0, consonant = 0;
      char prev;
      scanf("%s", str);
      if(strcmp(str, "end") == 0){
         break;
      }

      for(int i = 0; i < strlen(str); i++){
         if(str[i] == 'o' || str[i] == 'e' || str[i] == 'i' || str[i] == 'a' || str[i] == 'u'){
            flag = 1;
            vowel += 1;
            consonant = 0;
         }
         else{
            consonant += 1;
            vowel = 0;
         }
         if(i != 0 && str[i] != 'e' && str[i] != 'o'){
            if(str[i] == prev){
               flag = 0;
               break;
            }
         }
         if(vowel >= 3 || consonant >= 3){
            flag = 0;
            break;
         }

         prev = str[i];
      }

      if(flag){
         printf("<%s> is acceptable.\n", str);
      }
      else{
         printf("<%s> is not acceptable.\n", str);
      }
   }

   return 0;
}

File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.py
while(True):
  letter = input()
  if letter == 'end':
    break
  flag = False
  vowel, consonant = 0, 0

  for i in range(len(letter)):
    if letter[i] == 'a' or letter[i] == 'i' or letter[i] == 'o' or letter[i] == 'e' or letter[i] == 'u':
      vowel += 1
      consonant = 0
      flag = True
    else:
      consonant += 1
      vowel = 0
    
    if i > 0 and letter[i] == letter[i - 1] and letter[i] not in 'o' and letter[i] not in 'e':
      flag = False
      break
  
    if vowel >= 3 or consonant >= 3:
      flag = False
      break

  if flag:
    print(f"<{letter}> is acceptable.")
  else:
    print(f"<{letter}> is not acceptable.")